src/resources/ble_esp32.cc

```diff
@@ -227,12 +221,26 @@ bool TokenResourceMap::resize(int new_capacity) {
 class BleResourceGroup : public ResourceGroup {
  public:
   TAG(BleResourceGroup);
-  BleResourceGroup(Process* process, BleEventSource* event_source)
-      : ResourceGroup(process, event_source) {}
+  BleResourceGroup(Process* process, BleEventSource* event_source, Mutex* mutex)
+      : ResourceGroup(process, event_source)
+      , mutex_(mutex) {}
+
+  ~BleResourceGroup() override {
+    OS::dispose(mutex_);
+  }
 
   uint32_t on_event(Resource* resource, word data, uint32_t state) override;
 
   TokenResourceMap token_resource_map;
+
+  Mutex* mutex() {
+    return mutex_;
+  }
+
+ private:
+  /// A mutex protecting BLE operations.
+  /// NimBLE has its own thread, and we use this mutex to coordinate operations.
+  Mutex* mutex_ = null;
```

Comment:
Drop the initialization.

```suggestion
  Mutex* mutex_;
```

Reply:
done.